Library/Homebrew/cask/lib/hbc/audit.rb

```diff
@@ -243,6 +244,78 @@ def core_formula_url
       "#{core_tap.default_remote}/blob/master/Formula/#{cask.token}.rb"
     end
 
+    def check_https_availability
+      check_url_for_https_availability(cask.homepage) unless cask.url.to_s.empty?
+      check_url_for_https_availability(cask.appcast) unless cask.appcast.to_s.empty?
+      check_url_for_https_availability(cask.homepage) unless cask.homepage.to_s.empty?
+    end
+
+    def check_url_for_https_availability(url_to_check)
+      if schema_http?(url_to_check)
+        result, effective_url = access_url(url_to_check.sub(/^http:/, 'https:'))
+        if schema_https?(effective_url) && result == 1
+          add_error "Change #{url_to_check} to #{url_to_check.sub(/^http:/, 'https:')}"
+        else
+          result, effective_url = access_url(url_to_check)
+
+          if result == 0
+            add_error "URL is not reachable #{url_to_check}"
+          end
+        end
+      else
+        result, effective_url = access_url(url_to_check)
+        if result == 1 && schema_https?(effective_url)
+          return
+        else
+          result, effective_url = access_url(url_to_check.sub(/^https:/, 'http:'))
+          if result == 1 && schema_http?(effective_url)
+            add_error "Change #{url_to_check} to #{url_to_check.sub(/^https:/, 'http:')}"
+          else
+            add_error "URL is not reachable #{url_to_check}"
+          end
+        end
+      end
+    end
+
+    def access_url(url_to_access)
+      # return values:
+      #   1, effective URL : URL reachable, no schema change
+      #   0, nil           : URL unreachable
+      #  -1, effective URL : URL reachable, but schema changed
+
+      curl_executable, *args = curl_args(
+        "--compressed", "--location", "--fail",
+        "--write-out", "%{http_code} %{url_effective}",
+        "--output", "/dev/null",
+        url_to_access,
+        user_agent: :fake
+      )
```

Comment:
This will download the whole file to `/dev/null`, won't it? It should use `--head`.